.github/workflows/build-test.yml

```diff
@@ -654,14 +703,34 @@ jobs:
         crate: cargo-make
         version: latest
 
+    - name: Get twiggy version
+      id: twiggy-version
+      run: |
+        echo "::set-output name=hash::$(cargo search twiggy | grep '^twiggy =' | md5sum)"
+      shell: bash
+    - name: Attempt to load cached twiggy
+      uses: actions/cache@v2
+      id: twiggy-cache
+      with:
+        path: |
+          ~/.cargo/bin/twiggy
+          ~/.cargo/bin/twiggy.exe
+        key: ${{ runner.os }}-${{ steps.twiggy-version.outputs.hash }}
```

Comment:
Question (here and above): the [example](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action) has the key take the form

`${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}`

Your key is only

`${{ runner.os }}-${{ steps.twiggy-version.outputs.hash }}`

Do you need the middle two tokens ("build" and "env.cache-name") ?

Reply:
nah, because the git hash is already uniquely identifying

Reply:
I included the tool name in the key name for better debuggability